dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/teradata/TeradataQueryLogsJdbcTask.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright 2022-2024 Google LLC
+ * Copyright 2013-2021 CompilerWorks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.edwmigration.dumper.application.dumper.connector.teradata;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+import com.google.common.io.ByteSink;
+import com.google.edwmigration.dumper.application.dumper.handle.JdbcHandle;
+import com.google.edwmigration.dumper.application.dumper.task.AbstractJdbcTask;
+import com.google.edwmigration.dumper.application.dumper.task.TaskRunContext;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.StringWriter;
+import java.io.Writer;
+import java.sql.Clob;
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Base64;
+import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import org.apache.commons.io.IOUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.jdbc.core.ResultSetExtractor;
+
+public class TeradataQueryLogsJdbcTask extends AbstractJdbcTask<Void> {
+
+  private static final Logger LOG = LoggerFactory.getLogger(TeradataQueryLogsJdbcTask.class);
+  private final DateTimeFormatter SQL_FORMAT =
+      DateTimeFormatter.ISO_OFFSET_DATE_TIME.withZone(ZoneOffset.UTC);
+  private ZonedDateTime queryLogStartDate, queryLogEndDate;
+  private final String tableName;
+
+  public TeradataQueryLogsJdbcTask(
+      String targetPath,
+      String tableName,
+      ZonedDateTime queryLogStartDate,
+      ZonedDateTime queryLogEndDate) {
+    super(targetPath);
+    this.tableName = tableName;
+    this.queryLogStartDate = queryLogStartDate;
+    this.queryLogEndDate = queryLogEndDate;
+  }
+
+  @CheckForNull
+  @Override
+  protected Void doInConnection(
+      @Nonnull TaskRunContext context,
+      @Nonnull JdbcHandle jdbcHandle,
+      @Nonnull ByteSink sink,
+      @Nonnull Connection connection)
+      throws SQLException {
+    LOG.info("Getting first and last query log entries");
+
+    String[] queryLogsFirstAndLastEntries = new String[2];
```

Comment:
The information that this has to be a 2-element array is lost when this is passed as an argument to a method. The information that index=0 is the first entry and index=1 is the last isn't documented.

Please create a class for storing the first and last date.